src/main/java/gyro/aws/autoscaling/AutoScalingPlanResource.java

```diff
@@ -0,0 +1,286 @@
+    }
+
+    public void setCreationTime(String creationTime) {
+        this.creationTime = creationTime;
+    }
+
+    /**
+     * A list of scaling instruction configuration.
+     *
+     * @subresource gyro.aws.autoscaling.AutoScalingScalingInstruction
+     */
+    @Updatable
+    @Required
+    public List<AutoScalingScalingInstruction> getScalingInstructions() {
+        return scalingInstructions;
+    }
+
+    public void setScalingInstructions(List<AutoScalingScalingInstruction> scalingInstructions) {
+        this.scalingInstructions = scalingInstructions;
+    }
+
+    /**
+     * The name of the scaling plan.
+     */
+    @Id
+    @Required
+    public String getScalingPlanName() {
+        return scalingPlanName;
+    }
+
+    public void setScalingPlanName(String scalingPlanName) {
+        this.scalingPlanName = scalingPlanName;
+    }
+
+    /**
+     * The version of the scaling plan.
+     */
+    public Long getScalingPlanVersion() {
+        return scalingPlanVersion;
+    }
+
+    public void setScalingPlanVersion(Long scalingPlanVersion) {
+        this.scalingPlanVersion = scalingPlanVersion;
+    }
+
+    /**
+     * The status of the scaling plan.
+     */
+    @Output
+    @ValidStrings({
+        "Active",
+        "ActiveWithProblems",
+        "CreationInProgress",
+        "CreationFailed",
+        "DeletionInProgress",
+        "DeletionFailed",
+        "UpdateInProgress",
+        "UpdateFailed" })
+    public ScalingPlanStatusCode getStatusCode() {
+        return statusCode;
+    }
+
+    public void setStatusCode(ScalingPlanStatusCode statusCode) {
+        this.statusCode = statusCode;
+    }
+
+    /**
+     * The current status of the scaling plan.
+     */
+    @Output
+    public String getStatusMessage() {
+        return statusMessage;
+    }
+
+    public void setStatusMessage(String statusMessage) {
+        this.statusMessage = statusMessage;
+    }
+
+    /**
+     * The time and date when the scaling plan entered the current status.
+     */
+    @Output
+    public String getStatusStartTime() {
+        return statusStartTime;
+    }
+
+    public void setStatusStartTime(String statusStartTime) {
+        this.statusStartTime = statusStartTime;
+    }
+
+    @Override
+    public void copyFrom(ScalingPlan model) {
+        setCreationTime(model.creationTime().toString());
+        setScalingPlanName(model.scalingPlanName());
+        setScalingPlanVersion(model.scalingPlanVersion());
+        setStatusCode(model.statusCode());
+        setStatusMessage(model.statusMessage());
+        setStatusStartTime(model.statusStartTime().toString());
+
+        if (model.applicationSource() != null) {
+            AutoScalingApplicationSource applicationSource = newSubresource(AutoScalingApplicationSource.class);
+            applicationSource.copyFrom(model.applicationSource());
+            setApplicationSource(applicationSource);
+        } else {
+            setApplicationSource(null);
+        }
```

Comment:
For consistency with the lower block
```suggestion
        setApplicationSource(null);
        if (model.applicationSource() != null) {
            AutoScalingApplicationSource applicationSource = newSubresource(AutoScalingApplicationSource.class);
            applicationSource.copyFrom(model.applicationSource());
            setApplicationSource(applicationSource);
        } 
```